src/demoserver.cpp

```diff
@@ -264,22 +264,26 @@ void DemoServer::playback()
     if (current_packet.startsWith("MS#"))
       elapsed_time = 0;
 
-    while (!current_packet.startsWith("wait") && !demo_data.isEmpty()) {
+    while (!current_packet.startsWith("wait#")) {
         client_sock->write(current_packet.toUtf8());
+        if (demo_data.isEmpty())
+          break;
         current_packet = demo_data.dequeue();
```

Comment:
Ah crap how did I miss this, I thought I literally looked at the main playback loop